engine/orchestration/src/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -1763,7 +1763,7 @@ private void orchestrateStorageMigration(final String vmUuid, final StoragePool
             if (migrationResult) {
                 //if the vm is migrated to different pod in basic mode, need to reallocate ip
 
-                if (!vm.getPodIdToDeployIn().equals(destPool.getPodId())) {
+                if (destPool.getPodId() != null && !destPool.getPodId().equals(vm.getPodIdToDeployIn())) {
```

Comment:
destpool is not guaranteed to be not null either. Overlooked that detail before but maybe we want to change the condition to
(destPool != null && destPool.getPodId() != null && !destPool.getPodId().equals(vm.getPodIdToDeployIn()))